From 816e5885e7ac17e51f738e5310ab778bad9a22e1 Mon Sep 17 00:00:00 2001
From: Nicolas Ontiveros <niontive@microsoft.com>
Date: Mon, 1 Feb 2021 09:30:42 -0800
Subject: [PATCH] Disable tcrypt check in dracut-fips

Disable kernel crypto testing in dracut-fips. This codepath is necessary
for FIPS certified kernels and will be re-enabled when Mariner uses
a FIPS certified kernel.

---
 modules.d/01fips/module-setup.sh | 2 +-
 modules.d/01fips/fips.sh         | 4 ----
 2 files changed, 1 insertion(+), 5 deletions(-)

diff --git a/modules.d/01fips/module-setup.sh b/modules.d/01fips/module-setup.sh
index f011f15d..18af3a2a 100755
--- a/modules.d/01fips/module-setup.sh
+++ b/modules.d/01fips/module-setup.sh
@@ -36,7 +36,7 @@
         _fipsmodules+="ansi_cprng "

         # Misc:
-        _fipsmodules+="aead cryptomgr tcrypt crypto_user "
+        _fipsmodules+="aead cryptomgr crypto_user "
     fi

     mkdir -m 0755 -p "${initdir}/etc/modprobe.d"

diff --git a/modules.d/01fips/fips.sh b/modules.d/01fips/fips.sh
index 03da6861..958e07ed 100755
--- a/modules.d/01fips/fips.sh
+++ b/modules.d/01fips/fips.sh
@@ -99,10 +99,6 @@
     done
     mv /etc/modprobe.d/fips.conf.bak /etc/modprobe.d/fips.conf

-    info "Self testing crypto algorithms"
-    modprobe tcrypt || return 1
-    rmmod tcrypt
-
     info "Checking integrity of kernel"
     if [ -e "/run/initramfs/live/vmlinuz0" ]; then
         do_rhevh_check /run/initramfs/live/vmlinuz0 || return 1
 